### PR TITLE
OTE_SDK. Increase code coverage.

### PR DESCRIPTION
**Pull request by dvkruchinin**
Test list:
ote_sdk/ote_sdk/tests/entities/test_color.py
ote_sdk/ote_sdk/tests/entities/test_label.py
ote_sdk/ote_sdk/tests/entities/test_resultset.py
ote_sdk/ote_sdk/tests/entities/test_scored_label.py
ote_sdk/ote_sdk/tests/entities/test_task_environment.py

Task: CVS-66077

### CODE REVIEW

**Comment by dvkruchinin** [comment]
Hi @LeonidBeynenson,
please review the PR.

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

File: ote_sdk/ote_sdk/tests/entities/test_color.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestColorEntity:
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_color_entity(self):
+        """
+        <b>Description:</b>
+        Check that ColorEntity rizes some methods
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_color.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_color.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        <b>Expected results:</b>
+        Test passes if the rise NotImplementedError
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_color.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        assert self.car.hotkey == label_car_new_hotkey
+        assert self.car.id == label_car_new_id
+
+        test_label_entity_repr = [
+            f"{self.car.id}",
+            f"name={self.car.name}",
+            f"hotkey={self.car.hotkey}",
+            f"domain={self.car.domain}",
+            f"color={self.car.color}"
+
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        assert self.empty.id == ID()
+        assert type(self.empty.color) == Color
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_label_comparison(self):
+        """
+        <b>Description:</b>
+        Check that LabelEntity comparison
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Compare __lt__, __gt__ methods.
Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+        test_result_set_repr = [
+            f"model={result_set.model}",
+            f"ground_truth_dataset={result_set.ground_truth_dataset}",
+            f"prediction_dataset={result_set.prediction_dataset}",
+            f"purpose={result_set.purpose}",
+            f"performance={result_set.performance}",
+            f"creation_date={result_set.creation_date}",
+            f"id={result_set.id}"
+
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            "purpose": None,
+            "performance": None,
+            "creation_date": None,
+            "id": None,
+        }
+
+        result_set = ResultSetEntity(**test_data)
+
+        for name, value in test_data.items():
+            name = name[0]
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py:87]: Unfortunately, I don't understand the comment. Could you clarify?

File: ote_sdk/ote_sdk/tests/entities/test_scored_label.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    def test_scored_label(self):
+        """
+        <b>Description:</b>
+        Check that ScoredLabel can correctly return the value
+
+        <b>Input data:</b>
+        LabelEntity
+
+        <b>Expected results:</b>
+        Test passes if the results matches
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_scored_label.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+        Check that TaskEnvironment can correctly return the value
+
+        <b>Input data:</b>
+        Dummy data
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Using an already created dammy environment.
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_task_environment.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+        Check matches ConfigExample and dummy_config.yaml num_workers default_value
+
+        <b>Input data:</b>
+        ConfigExample, dummy_config.yaml
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Cheking num_workers default_value from ConfigExample and dummy_config.yaml
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_task_environment.py]: Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Ilya-Krylov) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py:87]: why do you take the first element from the name? I guess it was "model" and becomes "m".

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+        Check dummy_config missing_param
+
+        <b>Input data:</b>
+        dummy_config
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Cheking missing_param dummy_config
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_task_environment.py:271]: could you find all similar typos and fix?

File: ote_sdk/ote_sdk/tests/entities/test_color.py [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_color(self):
+        """
+        <b>Description:</b>
+        Check that Color can correctly return the value
+
+        <b>Expected results:</b>
+        Test passes if the results matches
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_color.py:46]: `results match`, please find all such entries and fix

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py]: Oh, thanks. Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py (hunk 10 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_task_environment.py]: Typos fixed

File: ote_sdk/ote_sdk/tests/entities/test_color.py (hunk 11 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_color.py]: Typos fixed

File: ote_sdk/ote_sdk/tests/entities/test_color.py [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestColorEntity:
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_color_entity(self):
+        """
+        <b>Description:</b>
+        Check that ColorEntity raises some methods
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_color.py:75]: raises some exceptions?

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+        assert self.empty.id == ID()
+        assert type(self.empty.color) == Color
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_label_comparison(self):
+        """
+        <b>Description:</b>
+        Check that LabelEntity __lt__, __gt__ methods with changed id
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_label.py:158]: ~that~

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/entities/test_color.py (hunk 12 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_color.py]: Sure. Fixed.

File: ote_sdk/ote_sdk/tests/entities/test_label.py (hunk 13 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done.

**Comment by Ilya-Krylov** [comment]
please have a look at issues found by CI  and fix them as well

**Comment by dvkruchinin** [comment]
> please have a look at issues found by CI and fix them as well

Fixed Pylint issues.

**Comment by dvkruchinin** [comment]
run ote_sdk tests

**Comment by dvkruchinin** [comment]
run ote_sdk tests

**Comment by dvkruchinin** [comment]
run ote_sdk tests

**Comment by dvkruchinin** [comment]
run ote_sdk tests

**Review by Ilya-Krylov — APPROVED** [review]

**Comment by Ilya-Krylov** [comment]
please fix test and re-create PR 
http://213.221.44.203/job/ote/job/precommit/job/trigger_ote_sdk/138/

**Pull request merged**